### ISSUE TITLE
Elemental block can't be archived if canDelete is false

### ISSUE THREAD

**Issue by Jianbinzhu** [silverstripe/silverstripe-elemental#856]
Hi,

Archiving on elemental blocks when `canDelete()` is return false gives the following error:

```
{data: {deleteBlock: null},…}
data: {deleteBlock: null}
errors: [{message: "Cannot delete DNADesign\Elemental\Models\BaseElement with ID 128",…}]
```

Also, adding `canArchive()` return false doesn't remove the archive functionality of the elemental.

![Screen Shot 2020-12-02 at 6 38 58 PM](https://user-images.githubusercontent.com/11606683/100832991-d7e54900-34cd-11eb-8824-fd7e6a921f9f.png)

### PR
- [x] https://github.com/dnadesign/silverstripe-elemental/pull/874

**Comment by emteknetnz** [comment]
>> Archiving on elemental blocks when canDelete() is return false gives the following error:

Archiving an element will delete the element, so this seems like the correct behavior

>> Also, adding canArchive() return false doesn't remove the archive functionality of the elemental.

Seems like that would make sense from a UX perspective.  What's the user experience when you select Archive when canArchive = false?

**Comment by scott1702** [comment]
> What's the user experience when you select Archive when canArchive = false?

I think we ignore canArchive as BaseElement uses canDelete; When canDelete = false the user receives no feedback when clicking archive, it just silently fails. Nothing happens to the block (Since the backend permissions kick in to not delete the element). 

Seems like the issue here is that the block actions don't respect the user's current permissions, they are all always shown regardless of a user's ability to action them. This issue isn't just about archive, It's the same for Unpublish, publish and assumedly duplicate as well (once https://github.com/dnadesign/silverstripe-elemental/issues/857 is fixed)

Note: Publish and unpublish actions do provide the user feedback

![image](https://user-images.githubusercontent.com/10215604/101303037-07b39880-38a2-11eb-9bfb-8a1417161316.png)

**Comment by emteknetnz** [comment]
Fixed in https://github.com/dnadesign/silverstripe-elemental/pull/874

**Comment by dnsl48** [comment]
The fix will be released in the following minor - 4.6.0
